doc/nrf/app_dev/programming.rst

```diff
@@ -56,16 +56,37 @@ Programming without ``--erase``
 
      west flash
 
+  ..
+
+.. _readback_protection_error:
+
 Programming with ``--recover``
-  Several Nordic Semiconductor SoCs or SiPs supported in the |NCS| offer an implementation of the :term:`Access port protection mechanism (AP-Protect)`, a form of readback protection.
-  To disable the AP-Protect, you must recover your device.
-  This is particularly important for multi-core devices.
+  Recovering a device typically refers to the process of resetting it to a known, working state, often by erasing and reprogramming its memory.
+  This is usually done in the following cases:
+
+  * The device is not functioning correctly.
+  * The device is using a readback protection, such as the :term:`Access port protection mechanism (AP-Protect)`, offered by several Nordic Semiconductor SoCs or SiPs supported in the |NCS|.
+    In such case, you might get an error similar to the following message:
+
+    .. code-block:: console
+
+       ERROR: The operation attempted is unavailable due to readback protection in
+       ERROR: your device. Please use --recover to unlock the device.
 
-  Use the following command:
+    Disabling the AP-Protect by recovering the device is particularly important for devices of the nRF53 and nRF54L Series.
+
+  Use the following command to recover your device:
 
   .. code-block:: console
 
      west flash --recover
 
-  This command uses ``nrfjprog --recover`` command in the background.
-  It erases all user available non-volatile memory and disables the readback protection mechanism if enabled.
+  This command erases all user-available non-volatile memory and disables the readback protection mechanism, if enabled.
+
+  .. toggle:: Background recovery commands
+
+     The ``west flash --recover`` command uses one of the following background commands:
+
+     * ``nrfjprog --recover`` for devices from nRF52, nRF70, and nRF91 Series.
+     * ``nrfutil device recover`` for devices from nRF53, nRF54H, and nRF54L Series.
+       For more information about how ``nrfutil device recover`` works, see the `nRF Util documentation <Recovering the device_>`_.
```

Comment:
This is more in vestawind/west developer domain, but since this is an internal implementation detail that could change, is the routing for different devices to nrfutil-device or nrfjprog something that should be documented here in the first place?

Reply:
Checked with @nordicjm : there is value in having this routing documented for users who want to build their own scripts that don't use west.